### PR TITLE
Automation account refactoring

### LINKED ISSUE #336: Update Automation Account to new child-resource structure Feature Request
## Description

Update Automation Account to new child-resource structure

### PR DESCRIPTION
**Pull request by eriqua**
# Change

Closes #336 

Pipeline reference: 
[Automation: Automationaccounts](https://github.com/Azure/ResourceModules/actions/workflows/ms.automation.automationaccounts.yml)

- Restructuring Automation Account module to include Sw Update Configurations
- Including deployment of Sw Update Configurations prerequisites such as
   - Link to log analytics workspace
   - _Updates_ solutions to the linked log analytics workspace
- Adding new feature for _choice_ and _boolean_ workflow inputs
   image

- Deleting old child resource pipeline and templates

**Note**: sw updates and private endpoint parameters have been tested but temporarily commented out. Custom removal job is needed in case they are deployed. See bug issue 517

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: arm/Microsoft.Automation/automationAccounts/softwareUpdateConfigurations/deploy.bicep [hunk 1]
```diff
@@ -174,24 +174,32 @@ param nextRunOffsetMinutes int = 0
...
 }
 
+resource softwareUpdateConfiguration_automationAccount 'Microsoft.Automation/automationAccounts@2020-01-13-preview' existing = {
+  name: automationAccountName
+}
+
 resource softwareUpdateConfiguration 'Microsoft.Automation/automationAccounts/softwareUpdateConfigurations@2019-06-01' = {
-  name: '${automationAccountName}/${deploymentScheduleName}'
+  name: name
+  parent: softwareUpdateConfiguration_automationAccount
```
Comment (MrMCake) [inline arm/Microsoft.Automation/automationAccounts/softwareUpdateConfigurations/deploy.bicep:202]: I know you're still in a draft mode, but as I'm just going through the entire repo on my search of non-aligned child-resources -> here is one :D

File: arm/Microsoft.Automation/automationAccounts/deploy.bicep [hunk 2]
```diff
@@ -172,6 +181,80 @@ module automationAccount_jobSchedules './jobSchedules/deploy.bicep' = [for (jobS
...
+  }
+  // This is to support solution to law in different subscription and resource group than the automation account.
+  // The current scope is used by default if no linked service is intended to be created.
+  scope: resourceGroup(!empty(linkedWorkspaceId) ? split(linkedWorkspaceId, '/')[2]: subscription().subscriptionId, !empty(linkedWorkspaceId) ? split(linkedWorkspaceId, '/')[4] : resourceGroup().name)
+  dependsOn: [
+    automationAccount_linkedService
+  ]
+}]
+
+module automationAccount_softwareUpdateConfigurations './softwareUpdateConfigurations/deploy.bicep' = [for (softwareUpdateConfiguration, index) in softwareUpdateConfigurations: {
```
Comment (MrMCake) [inline arm/Microsoft.Automation/automationAccounts/deploy.bicep:212]: note: leading `./` are redundant :)

Comment (eriqua) [reply arm/Microsoft.Automation/automationAccounts/deploy.bicep]: Cleaned up

File: .github/workflows/ms.automation.automationaccounts.yml [hunk 3]
```diff
@@ -4,17 +4,25 @@ on:
...
+        type: boolean
         description: 'Remove deployed module'
         required: false
         default: 'true'
       versioningOption:
-        description: 'The mode to handle the version increments [major|minor|patch]'
+        type: choice
+        description: 'The mode to handle the version increments [major|minor|patch|custom]'
         required: false
         default: 'patch'
```
Comment (MrMCake) [inline .github/workflows/ms.automation.automationaccounts.yml:15]: Why rolling back?

File: .github/workflows/ms.automation.automationaccounts.yml [hunk 4]
```diff
@@ -4,17 +4,25 @@ on:
...
+        options:
+          - major
+          - minor
+          - patch
+          - custom
```
Comment (MrMCake) [inline .github/workflows/ms.automation.automationaccounts.yml:20]: Should we do this only here or rather tackle this in a PR for all?

Comment (MrMCake) [reply .github/workflows/ms.automation.automationaccounts.yml:20]: The new inputs feature that is

File: .github/workflows/ms.automation.automationaccounts.yml (hunk 3 above)
Comment (MrMCake) [reply .github/workflows/ms.automation.automationaccounts.yml:15]: the newer is 'without' custom

File: .github/workflows/ms.automation.automationaccounts.yml (hunk 4 above)
Comment (MrMCake) [reply .github/workflows/ms.automation.automationaccounts.yml:20]: I'd suggest one search and replace and be gone with it. We should definitely get them in asap

File: .github/workflows/ms.automation.automationaccounts.yml (hunk 3 above)
Comment (eriqua) [reply .github/workflows/ms.automation.automationaccounts.yml:15]: fixed

File: .github/workflows/ms.automation.automationaccounts.yml (hunk 4 above)
Comment (eriqua) [reply .github/workflows/ms.automation.automationaccounts.yml]: I'll do that shortly after this PR is merged. I'd like to test the push to main

**Review by MrMCake — APPROVED** [review]

**Pull request merged**